cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-sync.php

```diff
@@ -119,66 +163,654 @@ public function is_synced( $post_id ) {
+					'priority' => 10,
+					'callback' => array( $this->managers['upload'], 'upload_asset' ),
+				),
+				'status'   => array(
+					'state' => 'info syncing',
+					'note'  => function () {
+						return sprintf( __( 'Copying to folder %s.', 'cloudinary' ), untrailingslashit( $this->managers['media']->get_cloudinary_folder() ) );
+					},
+				),
+			),
+			'public_id'   => array(
+				'generate' => array( $this->managers['media'], 'get_public_id' ),
+				'validate' => function ( $attachment_id ) {
+					$public_id = $this->managers['media']->has_public_id( $attachment_id );
+
+					return false === $public_id;
+				},
+				'sync'     => array(
+					'priority' => 20,
+					'callback' => array( $this->managers['media']->upgrade, 'convert_cloudinary_version' ), // Rename
+				),
+				'status'   => array(
+					'state' => 'info syncing',
+					'note'  => __( 'Updating metadata', 'cloudinary' ),
+				),
+			),
+			'suffix'      => array(
+				'generate' => array( $this, 'get_suffix_maybe' ),
+				'sync'     => array(
+					'priority' => 10,
+					'callback' => array( $this->managers['upload'], 'upload_asset' ),
+				),
+				'status'   => array(
+					'state' => 'uploading',
+					'note'  => __( 'Creating unique version', 'cloudinary' ),
+				),
+			),
+			'breakpoints' => array(
+				'generate' => array( $this->managers['media'], 'get_breakpoint_options' ),
+				'sync'     => array(
+					'priority' => 25,
+					'callback' => array( $this->managers['upload'], 'explicit_update' ),
+				),
+				'status'   => array(
+					'state' => 'info syncing',
+					'note'  => __( 'Updating breakpoints', 'cloudinary' ),
+				),
+			),
+			'options'     => array(
+				'generate' => array( $this->managers['media'], 'get_upload_options' ),
+				'sync'     => array(
+					'priority' => 30,
+					'callback' => array( $this->managers['upload'], 'context_update' ),
+				),
+				'status'   => array(
+					'state' => 'info syncing',
+					'note'  => __( 'Updating metadata', 'cloudinary' ),
+				),
+			),
+			'cloud_name'  => array(
+				'generate' => array( $this->managers['connect'], 'get_cloud_name' ),
+				'sync'     => array(
+					'priority' => 5,
+					'callback' => array( $this->managers['upload'], 'upload_asset' ),
+				),
+				'status'   => array(
+					'state' => 'uploading',
+					'note'  => __( 'Uploading to new cloud name.', 'cloudinary' ),
+				),
+			),
+		);
+
+		/**
+		 * Filter the sync base structure to allow other plugins to sync component callbacks.
+		 *
+		 * @param array $base_struct The base sync structure.
+		 *
+		 * @return array
+		 */
+		$base_struct = apply_filters( 'cloudinary_sync_base_struct', $base_struct );
+
+		// Apply a default to ensure parts exist.s
+		$structs = array_map(
+			function ( $component ) {
+				$sync_default   = array(
+					'priority' => 50,
+					'callback' => '__return_null',
+				);
+				$status_default = array(
+					'state' => 'sync',
+					'note'  => __( 'Syncronizing asset with Cloudinary', 'cloudinary' ),
```

Comment:
```suggestion
					'note'  => __( 'Synchronizing asset with Cloudinary', 'cloudinary' ),
```

Reply:
I would further that with `Synchronising` 😂